src/pyubx2/ubxtypes_get.py

```diff
@@ -3606,6 +3606,27 @@
         "response": U1,
         "reserved1": U3,
     },
+    "UPD-UPD1": {
+        "x1": "A12",
+    },
+    "UPD-UPD2": {},
+    "UPD-CHIPERASE": {},
+    "UPD-WRITE": {

```

Comment:
I suspect UPD-WRITE should actually be variable length  i.e. rather than:

```
    "UPD-WRITE": {
        "address": I4,
        "size": I4,
        "data": "A512",
    },
```

make it:

```
    "UPD-WRITE": {
        "address": I4,
        "size": I4,
        "groupdata": ( # repeating group * size
            "size",
            {
                "data": X1,
            },
        ),
    },
```

If i've misinterpreted the meaning of the size attribute and it doesn't represent the number of data bytes, then define the message as what u-blox call a 'variable by length' message - pyubx2 will work out the length automatically, i.e.

```
    "UPD-WRITE": {
        "address": I4,
        "size": I4,
        "groupdata": (
            "None",
            {
                "data": X1,
            },
        ),
    },
```

I suspect that your other changes will then not be needed.